x-pack/plugins/security_solution/public/timelines/store/timeline/epic_local_storage.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Action } from 'redux';
+import { map, filter, ignoreElements, tap, withLatestFrom, delay } from 'rxjs/operators';
+import { Epic } from 'redux-observable';
+import { get } from 'lodash/fp';
+
+import {
+  removeColumn,
+  upsertColumn,
+  applyDeltaToColumnWidth,
+  updateColumns,
+  updateItemsPerPage,
+  updateSort,
+} from './actions';
+import { TimelineEpicDependencies } from './types';
+import { isNotNull } from './helpers';
+
+const timelineActionTypes = [
+  removeColumn.type,
+  upsertColumn.type,
+  applyDeltaToColumnWidth.type,
+  updateColumns.type,
+  updateItemsPerPage.type,
+  updateSort.type,
+];
+
+export const isPageTimeline = (timelineId: string | undefined): boolean =>
+  // Is not a flyout timeline
+  !(timelineId && timelineId.toLowerCase().startsWith('timeline'));
+
+export const createTimelineLocalStorageEpic = <State>(): Epic<
+  Action,
+  Action,
+  State,
+  TimelineEpicDependencies<State>
+> => (action$, state$, { timelineByIdSelector, storage }) => {
+  const timeline$ = state$.pipe(map(timelineByIdSelector), filter(isNotNull));
+
+  return action$.pipe(
+    delay(500),
```

Comment:
maybe we should use `debounce` instead?

Reply:
I use `delay` here as we want to wait for the store to be updated first and then get the latest timeline. Is `debounce` better for this purpose?

Reply:
I tested with `debounceTime` and is not working as expected.